source/loader/layers/validation/ur_validation_layer.cpp

```diff
@@ -20,4 +22,125 @@ context_t::context_t() : logger(logger::create_logger("validation")) {}
 ///////////////////////////////////////////////////////////////////////////////
 context_t::~context_t() {}
 
+// Some adapters don't support all the queries yet, we should be lenient and
+// just not attempt to validate in those cases to preserve functionality.
+#define RETURN_ON_FAILURE(result)                                              \
+    if (result == UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION ||                   \
+        result == UR_RESULT_ERROR_UNSUPPORTED_FEATURE)                         \
+        return UR_RESULT_SUCCESS;                                              \
+    if (result != UR_RESULT_SUCCESS) {                                         \
+        assert(result == UR_RESULT_SUCCESS);                                   \
```

Comment:
I added this assert because I'm worried that error codes coming out of the GetInfo calls might make their way back to an application and be confusing. This way a debug build will quickly point out what's going wrong.

Reply:
sounds reasonable. maybe add `logger.error("unexpected return value {} from ...", result)`?